### PR TITLE
Update code base based on sp7

### PR DESCRIPTION
**Pull request by ahangarha**
### Summary

This PR updates the code base based on Shakapacker 7.

### Pull Request checklist

- [x] Add/update test to cover these changes
- [x] Update documentation
- [x] Update CHANGELOG file

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,10 +11,11 @@ _Please add entries here for your pull requests that are not yet released._
 
 ### Breaking Changes
 - Remove support & testing for Webpacker 3/4.
-- Add support & testing for Shakapacker v6
+- Add support & testing for Shakapacker
 - Requires separate compilations for server & client bundles (see [Webpack config](https://github.com/reactjs/react-rails/tree/master/test/dummy/config/webpack))
 - Replaces WebpackManifestContainer, which searched for assets in the webpack manifest, with SeparateServerBundleContainer, which expects a single server bundle file & does not use the webpack manifest at all.
 - All in one PR! #1274
```
Comment (justin808) [inline CHANGELOG.md:17]: improve clarity of formatting

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,10 +11,11 @@ _Please add entries here for your pull requests that are not yet released._
 
 ### Breaking Changes
```
Comment (justin808) [inline CHANGELOG.md:12]: Just require shakapacker v7 for v3 of react-rails so no older versions of webpacker, etc.

Comment (ahangarha) [reply CHANGELOG.md:12]: Exactly

**Comment by ahangarha** [comment]
Please check https://github.com/reactjs/react-rails/pull/1285/commits/9a2ec83f923bf9da4fadf258e71357bbd08674b1

That looks like a left-out debugging comment. I removed it, but please check if it was needed for something.

Also, take a look at https://github.com/reactjs/react-rails/pull/1285/commits/376c2fa1b88305c3a95efbd9f1b5c0bd0688b8f3.

**Review by Judahmeek — COMMENTED** [review]
test review

**Review by Judahmeek — APPROVED** [review]

**Review by Judahmeek — CHANGES_REQUESTED** [review]
test change request

**Review by Judahmeek — APPROVED** [review]

**Comment by ahangarha** [comment]
> test change request

@Judahmeek, May you be more specific?

**Comment by Judahmeek** [comment]
@ahangarha sorry. I was just experimenting with Github's review functionality.

**Pull request merged**